src/cache/impl/LevelDbCache.ts

```diff
@@ -22,19 +22,30 @@ export class LevelDbCache<V = any> implements SortKeyCache<V> {
    * and there doesn't seem to be any public interface/abstract type for all Level implementations
    * (the AbstractLevel is not exported from the package...)
    */
-  private readonly db: MemoryLevel;
+  private _db: MemoryLevel;
 
-  constructor(cacheOptions: CacheOptions) {
-    if (cacheOptions.inMemory) {
-      this.db = new MemoryLevel({ valueEncoding: 'json' });
-    } else {
-      if (!cacheOptions.dbLocation) {
-        throw new Error('LevelDb cache configuration error - no db location specified');
+  // Configuration kept to initialize the cache upon first access
+  private readonly cacheOptions: CacheOptions;
+
+  // Lazy initialization upon first access
+  private get db(): MemoryLevel {
+    if (!this._db) {
+      if (this.cacheOptions.inMemory) {
+        this._db = new MemoryLevel({ valueEncoding: 'json' });
+      } else {
+        if (!this.cacheOptions.dbLocation) {
+          throw new Error('LevelDb cache configuration error - no db location specified');
+        }
+        const dbLocation = this.cacheOptions.dbLocation;
+        this.logger.info(`Using location ${dbLocation}`);
+        this._db = new Level<string, any>(dbLocation, { valueEncoding: 'json' });
       }
-      const dbLocation = cacheOptions.dbLocation;
-      this.logger.info(`Using location ${dbLocation}`);
-      this.db = new Level<string, any>(dbLocation, { valueEncoding: 'json' });
     }
+    return this._db;
+  }
+
+  constructor(cacheOptions: CacheOptions) {
```

Comment:
since cache options is now a class field - `private readonly` can be added here, and line 28 and 48 removed.